lib/shared-sku-app/src/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,245 @@
+import * as React from 'react';
+
+import { AppExtensionSDK } from 'contentful-ui-extensions-sdk';
+import {
+  Heading,
+  Paragraph,
+  Typography,
+  TextField,
+  Form
+} from '@contentful/forma-36-react-components';
+import tokens from '@contentful/forma-36-tokens';
+import { css } from 'emotion';
+
+import FieldSelector from './FieldSelector';
+
+import { toInputParameters, toExtensionParameters } from './parameters';
+
+import {
+  getCompatibleFields,
+  currentStateToSelectedFields,
+  selectedFieldsToTargetState,
+  ContentType,
+  CompatibleFields,
+  SelectedFields
+} from './fields';
+
+import { Hash, ValidateParametersFn } from '../interfaces';
+
+interface Props {
+  sdk: AppExtensionSDK;
+  parameterDefinitions: Hash[];
+  validateParameters: ValidateParametersFn;
+  logo: string;
+  name: string;
+  color: string;
+  description: string;
+}
+
+interface State {
+  contentTypes: ContentType[];
+  compatibleFields: CompatibleFields;
+  selectedFields: SelectedFields;
+  parameters: Hash;
+}
+
+const styles = {
+  body: css({
+    height: 'auto',
+    minHeight: '65vh',
+    margin: '0 auto',
+    marginTop: tokens.spacingXl,
+    padding: `${tokens.spacingXl} ${tokens.spacing2Xl}`,
+    maxWidth: tokens.contentWidthText,
+    backgroundColor: tokens.colorWhite,
+    zIndex: 2,
+    boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.1)',
+    borderRadius: '2px'
+  }),
+  background: (color: string) =>
+    css({
+      display: 'block',
+      position: 'absolute',
+      zIndex: -1,
+      top: 0,
+      width: '100%',
+      height: '300px',
+      backgroundColor: color
+    }),
+  section: css({
+    margin: `${tokens.spacingXl} 0`
+  }),
+  splitter: css({
+    marginTop: tokens.spacingL,
+    marginBottom: tokens.spacingL,
+    border: 0,
+    height: '1px',
+    backgroundColor: tokens.colorElementMid
+  }),
+  icon: css({
+    display: 'flex',
+    justifyContent: 'center',
+    '> img': {
+      display: 'block',
+      width: '70px',
+      margin: `${tokens.spacingXl} 0`
+    }
+  })
+};
+
+export default class AppConfig extends React.Component<Props, State> {
+  state = {
+    contentTypes: [] as ContentType[],
+    compatibleFields: ({} as any) as CompatibleFields,
+    selectedFields: ({} as any) as SelectedFields,
+    parameters: toInputParameters(this.props.parameterDefinitions, null)
+  };
+
+  componentDidMount() {
+    this.init();
+  }
+
+  init = async () => {
+    const { space, platformAlpha } = this.props.sdk;
+
+    platformAlpha.app.onConfigure(this.onAppConfigure);
+
+    const [contentTypesResponse, currentState, parameters] = await Promise.all([
+      space.getContentTypes(),
+      platformAlpha.app.getCurrentState(),
+      platformAlpha.app.getParameters()
+    ]);
```

Comment:
We do not need any error handling here. If one of these promises doesn't resolve the `setReady` callback is never called and the app does not render. The web app errors out.

Reply:
Yeah, that seems like really bad UX to me. We should at least render the app and use the `notifier.error` or display a custom error message to the user. The `setReady` method not being called means that the entire app itself has failed to load due to other reasons.

If this code is running then we should provide the custom error messaging to the user since the app itself has not failed, only communications between services.